### PR TITLE
Feat(PE-3643): add create undername interaction

### PR DESCRIPTION
**Pull request by atticusofsparta**
image

note: currently the gateway is set to use ar-io.dev - this occasionally has issues on the gateway side with posting transactions, if so just change the gateway url in GlobalState.tsx to arweave.net, and it should work.

### CODE REVIEW

**Comment by dtfiedler** [comment]
rebase with develop so the diff shows only relevant changes

File: src/components/layout/Undernames/Undernames.tsx [hunk 1]
```diff
@@ -81,12 +116,100 @@ function Undernames() {
...
+      const confirmations = await arweaveDataProvider.getTransactionStatus(
+        new ArweaveTransactionID(id),
+      );
+      if (confirmations < 1) {
+        throw new Error(
+          'PDNT must have a minimum of 1 confirmation before editing, this protects against editing contracts lost due to block reorgs.',
+        );
+      }
+      switch (action) {
+        case UNDERNAME_TABLE_ACTIONS.CREATE:
```
Comment (dtfiedler) [inline src/components/layout/Undernames/Undernames.tsx:143]: i'm assuming this is a switch bc other actions logic will be added here?

Comment (atticusofsparta) [reply src/components/layout/Undernames/Undernames.tsx:143]: yup, edit and remove

File: src/components/layout/Undernames/Undernames.tsx [hunk 2]
```diff
@@ -112,6 +126,72 @@ function Undernames() {
...
   }
+  async function handleOnNext() {
+    try {
+      if (!id) {
+        throw new Error('No PDNT ID found, unable to perform transaction.');
+      }
+      const confirmations = await arweaveDataProvider.getTransactionStatus(
+        new ArweaveTransactionID(id),
+      );
+      if (confirmations < 1) {
```
Comment (dtfiedler) [inline src/components/layout/Undernames/Undernames.tsx:137]: Should this be something like 15?

**Review by dtfiedler — COMMENTED** [review]
I think we should set required confirmations to 15+ before editing. The table disables the manage button depending on the # of confirmations so we should match that.

File: src/components/layout/Undernames/Undernames.tsx (hunk 2 above)
Comment (atticusofsparta) [reply src/components/layout/Undernames/Undernames.tsx]: added

File: src/components/layout/Undernames/Undernames.tsx [hunk 3]
```diff
@@ -112,6 +131,70 @@ function Undernames() {
     setAction(undefined);
     setSelectedRow(undefined);
   }
+  function handleOnNext() {
+    try {
+      if (!id) {
+        throw new Error('No PDNT ID found, unable to perform transaction.');
+      }
+
+      if (confirmations < 15) {
```
Comment (dtfiedler) [inline src/components/layout/Undernames/Undernames.tsx:140]: nit: do we have a constant for this?

**Review by dtfiedler — APPROVED** [review]

**Pull request merged**